src/main/java/mate/academy/springintro/service/impl/UserServiceImpl.java

```diff
@@ -40,11 +54,90 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto)
 
         String roleName = Role.RoleName.USER.name();
         Role userRole = roleRepository.findByRole(Role.RoleName.USER)
-                .orElseThrow(() -> new EntityNotFoundException(String.format(NOT_FOUND_ROLE
-                        + roleName)));
-
+                .orElseThrow(() -> new EntityNotFoundException(String.format(NOT_FOUND_ROLE,
+                        roleName)));
         user.setRoles(Set.of(userRole));
         userRepository.save(user);
+        createShoppingCartForUser(user);
         return userMapper.toUserResponse(user);
     }
+
+    @Override
+    public ShoppingCartResponseDto getCartForUser(Long userId) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId)
+                .orElseThrow(() -> new RuntimeException("Shopping Cart not found for user "
+                        + userId));
+        return toShoppingCartDto(shoppingCart);
+    }
+
```

Comment:
Why do you move ShoppingCart logic to UserService? Let’s  move it to the separate ShoppingCartService class